test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangMath.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+package jit.test.recognizedMethod;
+import org.testng.AssertJUnit;
+import org.testng.annotations.Test;
+
+public class TestJavaLangMath {
+
+    /*
+    * The following test is for testing the java_lang_Math_sqrt function. 
+    * In this test, we are expecting the calling to the function to be transformed into a dsqrt node in recognizedCallTransformer.
+    * Also, if the input of the function is a double type constant, we should see the dsqrt node being reduced to dconst in treeSimplification.
+    * The above transformation and reduction will be performed when the function get executed twice, therefore, we set the invocationCount to 2.
+    * The test include the corner cases that sqrt requires in document.
+    */
```

Comment:
The documentation is a good start but it could use some improvements. We need to use Javadoc for Java documentation. We should also try to explain the _why_ and the _how_. For example the first sentence of the documentation is not very useful as this is quite obvious from the method name. The comment about the `iternationCount` is good though as it tells us some things that are definitely non-obvious.

Let's try to improve and format:

```
/**
 * Tests the constant corner cases defined by the {@link Math.sqrt} method.
 * <p>
 * The JIT compiler will transform calls to {@link Math.sqrt} within this test
 * into the following tree sequence:
 *
 * <code>
 * dsqrt
 *   dconst <x>
 * </code>
 *
 * Subsequent tree simplification passes will attempt to reduce this constant
 * operation to a <code>dsqrt<code> IL by performing the square root at compile
 * time. However we must ensure the result of the square root done by the
 * compiler at compile time will be exactly the same as the result had it been
 * done by the Java runtime at runtime. This test validates the results are the
 * same.
 */
```

Reply:
Similar change needed for the other file.